pkg/cloud/gke/storage/long_term_storage.go

```diff
@@ -40,5 +41,18 @@ func EnableLongTermStorage(installValues map[string]string, providedBucketName s
 		bucketName = bucketName[:60]
 	}
 
-	return createBucketFN(bucketName, "gs")
+	return createBucket(bucketName, installValues)
```

Comment:
I find the flow in this method confusing. Partly because of the two if statements and partly because of the multiple returns.

Reply:
I'd do something like:

```golang
func EnableLongTermStorage(installValues map[string]string, providedBucketName string) (string, error) {
	var bucketName string
	if providedBucketName != "" {
		bucketName = providedBucketName
		exists, err := gke.BucketExists(installValues[kube.ProjectID], bucketName)
		if err != nil {
			return "", errors.Wrap(err, "checking if the provided bucket exists")
		}
		if exists {
			return fmt.Sprintf("gs://%s", bucketName), nil
		}

		bucketURL, err := createBucket(bucketName, installValues)
		if err == nil {
			return bucketURL, nil
		}
		log.Warnf("Attempted to create the bucket %s in the project %s but failed, will now create a "+
			"random bucket", bucketName, installValues[kube.ProjectID])
         
                return createBucket(uniqueBucketName(installValues), installValues)
               
	} else 
		log.Info("No bucket name provided for long term storage, creating a new one")
                 return createBucket(uniqueBucketName(installValues), installValues)
	}
}
```

Reply:
Maybe even put the block in the first if statement into a function!?